### ISSUE TITLE
High power usage even when the app is not actively used

### ISSUE THREAD

**Issue by freijon** [wireapp/wire-android#853]
## The problem

I regularly get a warning from Android that Wire is "_draining my battery heavily_". When I compare the CPU total in the power overview, Wire seems unusual high compared to other apps.

## Environment

* Version: Wire 2.32.325 and older versions
* OS: OxygenOS OP3_O2_Open_15 (based on Android 7.1.1)
* Model: OnePlus 3
* Network: mostly on LTE

## Details

Open Wire. Wire doesn't even have do be used to drain the battery, it can be open in the background.

## Link to Debug logs

[wire_debug_report.txt](https://gist.github.com/freijon/244821e20ef5e426e23000bf12e31c9c)

**Comment by ec1oud** [comment]
Agreed, it's terrible.
![screenshot_2017-05-16-07-52-50](https://cloud.githubusercontent.com/assets/2946740/26091747/675891d8-3a0d-11e7-8114-e98b0134e6cc.png)

**Comment by kostadinstoilov** [comment]
I am also experiencing this issue. It started after upgrading from Wire 2.30.321 and has been present in all versions since. It seems like the issue is Wire keeps the device awake. See the attached screenshots for a comparison with Telegram on the same device.

![screenshot_20170528-141511](https://cloud.githubusercontent.com/assets/5895674/26528274/a3711576-43b0-11e7-9873-a12f3e67fb80.png)
![screenshot_20170528-141524](https://cloud.githubusercontent.com/assets/5895674/26528276/aceb7290-43b0-11e7-8bdc-6ffad20c6ede.png)

The device is running CM 13 which is based on Android 6.0.1 without Google Play Services.

**Comment by red-avtovo** [comment]
Hello everyone!
Is it happening with GCM enabled or not?

**Comment by freijon** [comment]
At the risk of sounding stupid: What is GCM? :)

**Comment by Frzifus** [comment]
I think he means [google#s push service](https://en.wikipedia.org/wiki/Google_Cloud_Messaging)

**Comment by kostadinstoilov** [comment]
Hello @red-avtovo 

I am observing this issue on a device without Google Play services so GCM is not being used.

**Comment by red-avtovo** [comment]
Basically we are using cloud based notification service system to be able to get notifications.
If we don't have ability to use GCM (Google Cloud Messaging = Google Play Services), the app is using WebSocket connection to the backend. That's why it drains quite much energy.

**Comment by kostadinstoilov** [comment]
That makes sense, but I did not experience this in Wire 2.30.321 so I believe there is some regression that is causing this battery drain issue.

**Comment by ec1oud** [comment]
Makes sense.

Sounds like there needs to be an open alternative for push notifications then.  What are all the other SMS-replacement apps doing?  I googled and I see that mqtt is one idea.  There must be a TCP connection always open, to the server, but it can be configured to be long-lived and not to require much maintenance traffic somehow.  For example https://ollieparsley.com/2013/05/20/using-mqtt-as-a-gcm-replacement-for-android-push-notifications/  http://tokudu.com/post/50024574938/how-to-implement-push-notifications-for-android  But then, that ought to be set up as a drop-in replacement solution and made available on F-Droid, so that every app doesn't have to open its own TCP connection.  (Searches there...) Oh look: https://sourceforge.net/projects/androidpn/   Also https://github.com/kontalk/androidclient/wiki/Push-notification-without-Google-Cloud-Messaging  https://microg.org/   That sounds like it would be worth a try.

Ideally the phone would have a routable IP address so that a central server wouldn't be required.  (The rollout of IPv6 was supposed to solve the address-shortage problem.)  Does that actually happen at least on some networks?  If it can be detected, maybe use that technique first, then fall back to MicroG on a phone which does not have GCM?

**Comment by kylgry** [comment]
I am also experiencing severe battery drain because of wire. It started with 2.34.338 or 2.33.336, I'm not sure. I was experiencing the problems described in issue #760 , which 2.33.336 update corrected. So I'm wondering if the latest update is not able to interface with GCM.  Keep awake for wire is basically the same as how long I've had the phone off the charger. 

I am using a google pixel running android 7.1.2 and google play services 11.0.55.

**Comment by iddno** [comment]
Oki i wrote my comment already here: https://github.com/wireapp/wire-android/issues/164
I don't know which one is the corect one now.
--------------------------------------------------------------------------------------------
I'm using latest WIRE version on Android, with a HTC 10 and Android 7.1.2 without gapps or any Google services tho! The Battery drain is massive and not normal or acceptable at all. I will deinstall this app until this is fixed. This app drains almost my entire battery life over one day, just alone, it is ridiculous.

http://imgur.com/a/QtLLA

I love WIRE and how the devs do work on it and keep it running and improving. But this has to be handled with priority imho.

Edit: I need to add, that Telegram does not have such battery issued at all, also Conversations.im XMPP app does not too. In fact, no other app drains that much battery like wire does currently. And I do not use google play services/GCM at all, on my device, so they all run without GCM actually.

**Comment by kostadinstoilov** [comment]
I don't believe this issue is related to GCM or WebSockets. I was using Wire with WebSockets for several months without any battery draining issues, my issues started after I upgraded from Wire 2.30.321. This was the last version that did not drain the battery for me. I would gladly downgrade to that APK as a workaround but it no longer works unfortunately

**Comment by klundry** [comment]
I am also seeing massive battery drain without gcm. I know wire will use more battery without google services btbi believe what we are experiencing is excessive and beyond what should be normal.  There has to be a bug.

I have also corresponded with another wire user that did not have nearly the battery drain some of us are seeing. This user was on version 2.28.317.

This issue is referring to the same bug affecting the users in issue #164.

**Comment by Awerick** [comment]
I had the same problem with two devices, one having [Google Play Services](https://play.google.com/store/apps/details?id=com.google.android.gms) (thus with GCM) but without a Google account, the other using [microG](https://microg.org/) (a free Google Services replacement also incl. GCM). Both devices were running fine before and the problem started as well with one of the recent Wire updates.
However **I could resolve the problem by manually upgrading Google Play Services resp. microG!**

**My speculation** is: The new Wire update expects a newer Google Play Services/GCM version and if GCM is not up-to-date anymore it silently falls back to WebSocket. As a result Wire is affected by the WebSocket bug (#164) that drains the battery.
If this speculation is correct, I would suggest that Wire should warn the user about the outdated GCM!
Also: It would be nice if Wire could show somewhere whether it actually uses GCM or WebSocket so that one can more easily identify this. (Maybe have it in the settings – in the future this could be extended to manual choose which one to use? :wink:)

**For those fellows using microG:**
_(For clarification: microG refers to a whole project. The app that replaces Google Services is called "Services Core" or "GmsCore")._
The last stable version of Services Core (0.2.4) uses an outdated version number – you need to update to one of the latest prerelease versions (0.2.4-…)!
Either download it directly from the [website](https://microg.org/download.html) (dropdown button reveals the prerelease versions) or update it via F-Droid (maybe "experimental updates" have to be enabled to see prerelease versions).
Unfortunately there is an [issue](https://github.com/microg/android_packages_apps_GmsCore/issues/190) with F-Droid and microG: If you have installed "Services Core" from the microG repository, it will show up as "Unified Nlp (no GAPPS)" from the official F-Droid repository. [This comment](https://github.com/microg/android_packages_apps_GmsCore/issues/271#issuecomment-279191150) by @landry314 explains ways how to update "Services Core" nevertheless.

**Comment by Larx** [comment]
I'm using the latest MicroG. However, there currently was an update to 0.2.4-103 (or something like that), and I have not yet noticed the battery drain (however for me it never was so bad as some report here).

**Comment by Strit** [comment]
I have a similar issue with mine.
The difference is that I get a notification about power draining, but when I go into details it uses far less then any other app in there, usually only 1%.

Wire (for android) version: 2.36.343 (from google play store.
Android OS: 7.1.1 (Ozygen OS 4.1.6)

**Comment by freijon** [comment]
@Strit I think this is because it drains the power via Google Play Services, at least that is how I think it works. Google Play Services power consumption is much higher if Wire is running.

I hope someone is going to work on this issue soon. I stopped using Wire on my mobile because of this.

**Comment by Strit** [comment]
Might be true @freijon but I don't notice any real battery drain at all. It still has 50% left on the battery when I go to bed.

**Comment by freijon** [comment]
For me it's 50% in the evening when Wire is running during the day vs. up to 3 days without Wire.

**Comment by Awerick** [comment]
@freijon: It would be interesting to know which "Google Play Services" version is installed on your phone.
The latest version seems to be `11.0.??`. I am not sure about version `10` but if you are on version `9` or below, this is probably the reason for the battery drain!
(Do you use the "Google Play Store" and let it automatically update your apps? If so, it should actually update "Google Play Services" automatically…)

**Comment by freijon** [comment]
@Awerick: I have version 11.0.55 installed on my device.

**Comment by Awerick** [comment]
And still the battery gets drained? Interesting, then there seems to be another issue besides(/instead?) the one I speculated…

**Comment by decuba** [comment]
Same issue on LineageOS without gapps. No microG eather. Battery drain ridiculously high. I downgraded to v2.30.321 as stated by @kostadinstoilov for a camarison. No battery drain, the version is buggy though. The issue obviously concerns both, GSM and WebSocket. Hope they can fix it soon.

**Comment by Nubzori** [comment]
Experiencing similiar drain on CopperheadOS, which means no gapps.

**Comment by freijon** [comment]
What I'm also not getting: Why is this labeled as "enhancement"? It's clearly an issue (bug?) and not a feature request. And it's a killer for me!

**Comment by CL4RSEN** [comment]
Any updates on this bug from the Wire team?

**Comment by lucilacaranza** [comment]
Hello @freijon, @CL4RSEN and everyone,

We are aware of the battery issue and it is marked 'high priority' on our end. We will most likely have a service run to fix bugs by the end of July.
Thanks for all the feedback you have provided!
Best,

**Comment by bradfier** [comment]
Hi,

In case it's any use to your team, it looks like the battery issue (in my case anyway, Galaxy S7, Android 7.0) is caused by the ZMessagingService in the sync library not releasing WakeLocks:

![screenshot_20170715-102140](https://user-images.githubusercontent.com/3495954/28238273-093e100e-6948-11e7-8ad2-9510aa72fab4.jpg)

**Comment by sergachev** [comment]
In the latest version (2.37.349), one wakelock (FutureService) remains out of two. 
Had both FutureService and ZMessagingService on the previous version of Wire.
Android 7.1.2, no google apps.

**Comment by decuba** [comment]
Battery drain still extremely high in the new version 2.37.349. Seems like nothing has chaged at this point. In my case it is 15% to 18% after one day run without any active use of wire-app (lineageOS, no gapps, OnePlus3). For me the app is useless untill it's fixed.

**Comment by joaobarcia** [comment]
The three friends I had on WIre had to uninstall because of this issue and I have been left alone with Otto the Bot. I am aware this is not exactly a technically constructive comment, but I hope it helps reinforce how high of a priority this issue should be. If you loose your user base because of basic instability, there is no point in developing fancy new features.

Happy to gather any info that might be relevant for debugging.

**Comment by bfqrst** [comment]
I couldn't agree more with @joaobarcia's comment. I am heavily evangelizing for Wire, but at times I'm close to uninstalling because of the enormous power drain. I forcefully stopped the app many times, to be able to preserve some battery charge, which completely defeats the purpose of a messenger... Likewise, I'm happy to provide helping input!

**Comment by sergachev** [comment]
I would add: the app _does_ sleep sometimes (i.e. during the night when the phone is not used at all - this plateau on the screenshot). It seems like the sleep starts after ~1 h of screen being off.

![screenshot_20170724-121533](https://user-images.githubusercontent.com/1894984/28518957-1fa5a978-706a-11e7-98f2-84acf9b07987.png)

**Comment by bfqrst** [comment]
Right @sergachev ! It is erratic. Sometimes after a full recharge it rockets straight to the top without even opening Wire a single time!!! Other times the "keep awake" stops a certain point... I haven't been able to come to a conclusion up to now.

**Comment by tmikaeld** [comment]
One of my clients just left Wire due to this issue alone, that's 7 people that stopped using it - making me look stupid for even recommending it in the first place. 

If i where you, I'd put all resources into finding and fixing this bug - It's been around so long that even those with a ton of patience are starting to loose it. (OnePlus 3 and Samsung Galaxy S7)

**Comment by decuba** [comment]
Once again the issue unsolved in the new version 2.38.352. I'm done. Moved to Riot/Matrix, 12 people.

**Comment by bfqrst** [comment]
Seriously, I'm not sure how much longer I'm willing to deal with this... I stop Wire forcefully way more often than it is online, which, as I stated before defeats the purpose of a messenger. And the fact that this has the enhancement label is ... quite of a misnomer!

**Comment by avently** [comment]
Yes, i removed it too. Messenger should be a messenger but not a battery drainer.

**Comment by nychtos** [comment]
Same Issue:
Nexus 6P - Android O DP4

**Comment by ganego** [comment]
[![bat.png](https://s3.postimg.org/7kbatdx8j/bat.png)](https://postimg.org/image/u90hsyem7/)

Wanna take a guess when I started Wire?
Huawei P6 @ Android 4.2.2. Google Play Services 11.3.02 (the app is only 32 MB according to the phone, but the one on my Redmi Note 4 is 257 MB?).

On my Redmi Note 4 (MIUI Global 8.2 (Android 6.0.1) no issues at all.

**Comment by freijon** [comment]
@decuba I didn't know about Riot / Matrix. I'm giving it a try now. Thanks for that info.

**Comment by hchr** [comment]
Pixel XL 7.1.2 with latest security updates

![wire-power-usage](https://user-images.githubusercontent.com/6417330/28981964-4684ceb4-7954-11e7-86f6-f97a0506fa28.png)

**Comment by someoneEsle** [comment]
Hi guys, a quick update. We have been working on this lately, and are going to release an update soon that should fix some of your problems. For those of you who were experiencing battery drains with GCM installed, things should get better. We fixed one of the potential causes of the battery drain, so please keep an eye on it.
Thank you for your feedback and understanding.

**Comment by decuba** [comment]
Is this gonna fix battery drain for systems without GCM/microG installed?

**Comment by someoneEsle** [comment]
It's early to say for sure, but yes, we think they should be affected too. :)

**Comment by hchr** [comment]
If you can't wait for Google/wire.com to update to Version 354, you can download the apk from apkmirror:
http://www.apkmirror.com/apk/wire-swiss-gmbh/wire/wire-2-39-354-release/

wire.com only has Version 352

**Comment by hchr** [comment]
![screenshot_20170810-114738 2](https://user-images.githubusercontent.com/6417330/29165225-54576994-7dc2-11e7-9d3c-cd1813ce5908.png)

After 2 hours, it's using a lot less power, thank you :-)

**Comment by someoneEsle** [comment]
@hchr Thanks for your report! I see you are on a Pixel, do you have stock android and settings? Do you use Gplay services, or do you not? Would be interesting for us to gather data about which phones are seeing improvements and which ones aren't. Thanks!

**Comment by hchr** [comment]
@someoneEsle 
>do you have stock android and settings?
Yes to stock Android. I have change some settings, however not many. If u want to know about some specific settings, let me know. 
>Do you use Gplay services
Yes. Version 11.3.02

**Comment by kostadinstoilov** [comment]
Unfortunately I am still getting this issue on the latest Version: 2.39.356 (August 15, 2017)

Running on a LineageOS 14.1 nightly without Google Play Services

The app keeps the device awake when not being used, deteriorating battery life.

![screenshot_20170816-140836](https://user-images.githubusercontent.com/5895674/29360684-c68ab11e-828c-11e7-987a-82237f1b5212.png)

**Comment by decuba** [comment]
Although I successfully moved to Riot/Matrix a couple of weeks ago, I still kept an eye on Wire. I'm on LineageOS w/o gapps, Wire version 2.39.356 (August 15, 2017). The battery consumption hasn't noticeably changed over the past several fixes on my device either. It still ridiculousely drains the baterry. I also tried to run with microG as Gms for pushing messages - same thing, huge battery drain. 

Keeping in mind that the (future-) funding of the app is very vague, plus this reallly long unsolved issue - I'm giving up. I completely removed Wire. Battery life's back to normal. Maybe I'll check it next year (if it's still not sold to Facebook, Goole or MS, just think on Skype).

**Comment by Awerick** [comment]
@decuba: Just a short remark: If you installed microG/GmsCore _after_ Wire, it is probably not used.
 >  "Apps that use Cloud Messaging must be installed after GmsCore, or else they will not work."
 >  [Source: [microG wiki](https://github.com/microg/android_packages_apps_GmsCore/wiki/Helpful-Information)]

**Comment by tmikaeld** [comment]
My battery have been normal again since after this update, but i did clear any App data so to reset the Wire data completely. Not sure if that helped, but at least now it works normally.

**Comment by decuba** [comment]
@Awerick: you're right, an app doesn't register to GSM if installed prior to microG. I cleared data, reinstalled Wire and made sure it registered appropriately in microG settings. It might have reduced battery consumption a bit, but barely noticeable, if at all. Wire was still on the top of the battery consumers list with very high percentage.

**Comment by kostadinstoilov** [comment]
I also reinstalled Wire, but that did not help at all.

**Comment by decuba** [comment]
It might depend on OS. Apparently no luck on custom roms (al least lineageos) without gapps.

**Comment by jussihuotari** [comment]
Just FYI, I'm still getting these "High power usage reminder - Wire is draining your batter" notifications.

My setup is off-the-shelf plain vanilla: 

* Wire 2.39.356
* Oxygen OS and Android 7.1.1
* Google Play Services 11.3.02

Hope this info helps in finding a fix for the problem. Wire is great.

**Comment by tmikaeld** [comment]
Today the Wire high power usage issue came back, does it draw more power when reception is bad? Because i was travelling today, I've had no issues whatsoever since the "fix", but today it occurred twice.

**Comment by someoneEsle** [comment]
@tmikaeld  what device are you on?

**Comment by tmikaeld** [comment]
@someoneEsle OnePlus 3, unrooted, latest OxygenOS.

**Comment by Strit** [comment]
Same for me. Also OnePlus 3, latest OxygenOS. it tells me it uses high amounts of power, but the battery drain is not present.

**Comment by Protinus3000** [comment]
Are there any news regarding this issue? I have the same problems. I'm using a Samsung S4 mini and Lineageos 14.1

**Comment by someoneEsle** [comment]
We have an idea of what it is, and if we are correct it is related to some notification issues that some people are experiencing. We are gathering a group of beta testers to test out our ideas in the coming days/weeks. See #1087

**Comment by blahmutz** [comment]
Is it a public beta or a closed beta? would love to join!

**Comment by someoneEsle** [comment]
@blahmutz of course you can join! See here https://github.com/wireapp/wire-android/issues/1087#issuecomment-334402971

**Comment by blahmutz** [comment]
@someoneEsle thats bad, i dont have play. I have nothing of google installed on that device. Should i still write an email to that address?

**Comment by DDvO** [comment]
More than 3 months after the last comment here, the problem persists.
I recently opened a new issue on it: #1388 to give it more visibility.

**Comment by DDvO** [comment]
@someoneEsle or who else is responsible here: 
Wire would be such a cool secure IM solution - but as long as this battery drain is not really fixed, users are driven away. 
Moreover, this issue is a sign of bad programming and a sign of bad software management - fixing it should be given very high priority, about as high as fixing security bugs.

**Comment by someoneEsle** [comment]
Hey, yes we hear you. Your feedback never goes unheard and we are constantly working on improving the app.

**Comment by klundry** [comment]
So can we get an update and some more info than just we hear you and are working on it? It's been a long time that this has been an issue so it would help quell some fears and keep people on the platform instead of just leaving like a lot of people are doing already.

**Comment by DDvO** [comment]
BTW, I don't think the problem is necessarily GCM related because on my phone with custom rooted Android 7 I do use the gapps like Google Play Services. Or maybe GCM is still disabled somehow; how can one check whether GCM is enabled and (re-)enable it if not?

Interestingly, after killing the Wire app as a manual workaround for the battery drain, the app still somehow is signaled and wakes up when a new message arrives. This is a further indication that the problem at least in my case is independent of (the availability of) GCM.

**Comment by blahmutz** [comment]
When i install gapps i did not have this heavy battery drain (iirc). I do not want to test because then i have to reflash,  because this gapps stuff is not uninstallable without reflashing. remembers me on malware.

i do not know how to check if your phone uses gcm except sending a message to your phone. afaik it is required to first install gapps and then wire. i dont know if that happend in your case. the only way i know to check if your phone uses gcm is to send you a message over gcm. 
maybe this can be used to check: https://github.com/alvareztech/firebase-ui-android-auth-phone

**Comment by kristin-wire** [comment]
Firstly, a big thank you to all of you who have contributed your comments and feedback regarding this issue. Wire developers read these reports first hand, and they are helpful in the troubleshooting process.
 
This specific issue is rather tricky, as it affects some users (like yourselves), and not others. Tracking it is on our constant radar, and your feedback often provides clues for us in this process. 

While bug fixes and feature developments are a part of each sprint, some issues - like this one- are more of a marathon than a sprint. What we are doing now is running optimization tests and mapping the variables contributing to this issue.

**Comment by Protinus3000** [comment]
@kristin-wire thanks for the update! 
@DDvO For me that sounds like your gcm is working. I'm using a lineageos 14.1 without gcm and when I kill the app, it doesn't wake up when a new message arrives. Because there is no messaging service (websocket or so) running.

**Comment by freijon** [comment]
@Protinus3000, maybe @DDvO is using microG?

**Comment by DDvO** [comment]
I must withdraw my statement that I receive messages when the Wire app has been killed. I thought it was so, but I've meanwhile done several tests and found that Wire does not really wake up when a new message arrives. Sorry for the confusion.

As I do use Google Play Services etc., I don't understand why apparently GCM is not active or does not work with Wire on my phone. Maybe this has to do with app permissions?

**Comment by DDvO** [comment]
**EDIT**
After the Wire app has been killed, it does not wake up to notify me when a copy of new message arrives that I sent (to someone else) from another device; it just displays my message after I've manually invoked Wire. 
When someone else sends me a message, I do get notified immediately even if the app has been killed before (without draining the battery in the meantime). 
Once the app has been opened, it starts again continuously wasting some 5% CPU.

I meanwhile tried using microG, but cannot install it on my phone.
Since I do use Google Play Services etc., I do not understand why apparently GCM does not work properly with Wire on my phone.

**Comment by DDvO** [comment]
@someoneEsle  and @kristin-wire, you wrote that you are working on the issue.
Do you meanwhile know what exactly causes the problem and 'just' need to fix and test it,
or are you still in the analysis stage? 

Does the app (still) rely on GCM, and if so, which version of Google Play Service should work? I have 11.9.75

**Comment by DDvO** [comment]
Note that other IM apps like Conversations made sure that they do nice to the battery even without GCM.

For instance, [https://conversations.im/](https://conversations.im/) states: _**Low impact on battery**_
_Even though Conversations keeps its own connection to the XMPP server and thus is independent of Google’s push messaging system (GCM), it does a lot of work to keep the impact on battery life as low as possible._ 

And indeed Conversations, Kandru, Threema, and WhatsApp use just around 0.3% CPU on my phone, and Signal even less.

**Comment by 80tf3262p8** [comment]
Echoing some others, I don't use Google Cloud Messaging (no Google Play Services) and I have this issue.

Signal (which maintains a constant connection of its own, in this situation) uses maybe about a 10th as much battery, even if I have actively used Signal but haven't used Wire at all.

DDvO, you asked earlier about checking if GCM is working. Here is one app that does that: https://play.google.com/store/apps/details?id=com.firstrowria.pushnotificationtester

**Comment by DDvO** [comment]
Thanks @80tf3262p8  for referring me to that app!

I just tried it out and found that GCM does work on my device.
So pretty weird that I still experience battery drain when Wire has been opened.

This app must have a more fundamental flaw leading to some sort of busy loop.

**Comment by tokariu** [comment]
I've reported this battery drain issue over a year ago while i was still using wire with websockets. After a month or two I decided to enable google frameworks again for GCM (I disabled it because I highly prefered a google-free smartphone) because with GCM there was no battery drain ( at least not as much as with websockets).
Now almost a year later for other reasons I disabled the google framework / GCM again and had Wire running over websockets again. 
And what should I say... Wire process uses between 58% and 73% each day of my battery in the first 3 days with websockets. I almost had no battery after only half of the day.
My Specs are still the same: LG G3, Android 6.0, rooted stock, latest public wire.

NOTHING has changed after over ONE YEAR of this abysmally battery drain issue! I'm more than disappointed to say the least.

I hate to use Wire with GCM again, but there is still no way to use it with websockets. This hurts so much because serious competitors have already developed working battery friendly solutions with websockets.

I'd really appreciate if the Wire team would focus on stability and basic problems again instead of all this team- and corporate features only (at least that's what it looks like to me nowadays :/ )

**Comment by Protinus3000** [comment]
Hey, are there any news regardig this issue? It's become silent in here...

**Comment by ra-dierer** [comment]
The problem is still present! I use LineageOS on two different devices without GApps or GCM and have the same problem on both devices with massive battery drain in idle state. Other messengers like Signal are working very well. In my opinion Wire is one of the best messenger, but with this battery drain bug the Android version became unusable.

Wire Version 3.15.632
LineageOS 41.1
Android 7.1.2

**Comment by ckquentvp** [comment]
I don't think I'm necessarily adding anything here by saying that I'm very much having the same issues as other people here.

The problem is worse on the newest version (at the time of writing), which has forced me to downgrade and left me wondering, whether Wire is a viable alternative to other similar platforms.

It boggles my mind that the usage can be 25%, when it has not been used.
https://user-images.githubusercontent.com/42350553/44041677-e63afb6a-9f0d-11e8-93d1-0ae7d841bdea.png

**Comment by sebaki** [comment]
I am experiencing the same issue on my Moto X Play using Lineage OS 14.1 (Android 7.1.2) without google services.
As this is a issue, which is regarding many users, is this going to be solved?

Cheers,
s

**Comment by JonasT** [comment]
Same problem on LineageOS 14.1 with S4 mini - when I open the app *once*, even for just a few seconds, and then never enter it again, it will still drain battery more than all other apps in the battery graph even though no single message is ever received, and I'm never opened up again after that very brief first time. That's not really how a correctly working Android messaging app is supposed to behave :confused:

**Comment by Protinus3000** [comment]
For me too! But awkwardly notification reliability has become worse, not better...

**Comment by GitStorageOne** [comment]
This is huge problem and still exists!  

![image](https://user-images.githubusercontent.com/3978064/135716714-86fbaf3e-a92b-490d-b307-f70f6ebec2d1.png)

**Comment by DDvO** [comment]
I suppose using so-called Long Polling would solve the issue.
